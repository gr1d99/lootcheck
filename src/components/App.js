import React from 'react';
import Wallet from "./Wallet";
import Loot from "./Loot";

class App extends React.Component {
	render() {
		return (
			<div>
				<h2>Loot Checker</h2>
				<hr/>
				<Wallet/>
				<hr />
				<Loot />
				<div>
					Powered by <a target="_blank" rel="noopener noreferrer"  href="https://www.coindesk.com/price">Coindesk</a>
				</div>
			</div>
		)
	}
}

App.displayName = 'App'

export default App;
