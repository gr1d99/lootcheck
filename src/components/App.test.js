import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import App from './App'

import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'


const mockStore = configureStore([thunk])
const store = mockStore(0)

const setup = () => (
	mount(
		<Provider store={store}>
			<App />
		</Provider>
	)
)


describe('<App />', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = setup()
	})

	it('renders correctly', () => {
		const tree = renderer.create(wrapper).toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('contains a connected <Connect(Wallet) /> component', () => {
		expect(wrapper.find('Connect(Wallet)').exists()).toBe(true);
	})

	it('contains a connected <Connect(Loot) /> component', () => {
		expect(wrapper.find('Connect(Loot)').exists()).toBe(true);
	})

	it('contains a link to the coindesk price page', () => {
		expect(wrapper.find('a').props().href).toBe('https://www.coindesk.com/price')
	})
})
