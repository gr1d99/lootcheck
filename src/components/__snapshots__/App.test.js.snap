// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div>
  <h2>
    Loot Checker
  </h2>
  <hr />
  <div>
    <h3
      className="balance"
    >
      Wallet Balance: 
    </h3>
    <input
      className="input-wallet"
      onChange={[Function]}
      type="number"
    />
    <button
      className="btn-deposit"
      onClick={[Function]}
    >
      Deposit
    </button>
    <button
      className="btn-withdraw"
      onClick={[Function]}
    >
      Withdraw
    </button>
  </div>
  <hr />
  <h3>
    Bitcoin Balance: 
    
  </h3>
  <div>
    Powered by 
    <a
      href="https://www.coindesk.com/price"
      rel="noopener noreferrer"
      target="_blank"
    >
      Coindesk
    </a>
  </div>
</div>
`;
